accord-core/src/main/java/accord/local/CommandStore.java

```diff
@@ -548,6 +563,34 @@ final Supplier<EpochReady> bootstrapper(Node node, Ranges newRanges, long epoch)
         };
     }
 
+    public boolean safeToRespond(TxnId txnId, Unseekables<?> participants)
+    {
+        return !unsafeGetRedundantBefore().isUnsafeBefore(txnId, participants);
+    }
+
+    protected EpochReady rebootstrap(Node node, Ranges ranges, long epoch)
+    {
+        AsyncResult<EpochReady> metadata = submit(empty(), safeStore -> {
+            Bootstrap bootstrap = new Bootstrap(node, this, epoch, ranges, DataStore.RequestKind.Sync);
+            bootstraps.add(bootstrap);
+            // If rebootstrap can grab a later timestamp for subsequent attempts, but this timestamp is enough for us
+            // to establish what's safe to read
+            TxnId unsafeBefore = bootstrap.start(safeStore);
+            logger.debug("Rebootstrap timestamp on {}@{}: {}", id, node.id(), unsafeBefore);
+            unsafeUpsertRedundantBefore(RedundantBefore.create(ranges, unsafeBefore, SHARD_UNSAFE_BEFORE));
```

Comment:
I think we need a safeStore here don't we? Else this won't be persisted, at least in the integration.